### DISCUSSION TITLE
2 Victron Geräte an einem ESP

### DISCUSSION THREAD

**Discussion by openKimchi** [softwarecrash/Victron2MQTT#20] [General]
Hallo,
kann man auch zwei Victron Geräte an einem ESP anschließen oder benötigt man für jedes Victron Gerät ein eigenes ESP?
viele Grüße

**Comment by softwarecrash** [comment]
Aktuell ist nur ein gerät möglich, es braucht für jeden einen esp

**Reply by hoschiking** [reply]
Tolles Projekt, aber mehr als 1 MPPT pro ESP wäre schon super. Ich habe 4 Laderegler, bräuchte also 4x ESP.
Ich denke, die die allermesiten haben mehr als einen MPTT ;-)
Das wäre eine schöne Verbesserugn des Projektes

**Reply by softwarecrash** [reply]
@hoschiking das wird aus mehreren gründen nicht umgesetzt
- Zusammenlegung der maßen, meist keine optimale Lösung
- VE.Direct ist ein 1-way Protokoll, der mppt sendet im Sekunden Takt seine daten, jetzt mehrere abzufragen würde dazu führen das der esp (1 core) damit beschäftigt ist eine Nachricht zu empfangen/verarbeiten während ein weiterer bereits sendet, so gehen Nachrichten verloren.
- dem esp gehen die pins aus, ja er hat viele; aber nur eine einige sind wirklich frei verwendbar ohne das es zu Boot Problemen führen kann.

**Reply by hoschiking** [reply]
Ja, all das ist bekannt und in anderen Projekten schafft ein ESP locker mehrere MPPT´s zu loggen, von daher
denke ich nicht, dass das ein Problem sein sollte. Meines Erachtens hat der ESP genug Rechenleistung dafür.
Aber ich bin auch kein Software-Experte, kann es daher nicht persönlich beurteilen.
